### PR TITLE
feat(ci): add release action

### LINKED ISSUE #68: add GitHub Action to deploy CDK app
## Objective

Create a GitHub Action to deploy CDK application.

- configure e2e Discord Bot secrets (defined in `.env.sample`)
- [configure AWS credentials](https://github.com/aws-actions/configure-aws-credentials)

## Steps

- run tests with `pnpm test`
- run library build with `pnpm build`
- run CDK synth with `cd cdk; pnpm cdk synth -c  env={main|next}; cd -`
- run CDK deploy with `cd cdk; pnpm cdk deploy -c env={main|next} --all; cd -`

### Considerations

- Docker build cache
- pnpm cache
- diff deploy CDK stacks based on app change?

### LINKED ISSUE #168: add cdk stack for GitHub Action setup with AWS

### PR DESCRIPTION
**Pull request by josefaidt**
*Issue #, if available:*

closes #68 
closes #168 

*Description of changes:*

- adds release action

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: cdk/src/app-github-actions.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+import * as iam from 'aws-cdk-lib/aws-iam'
+import { Construct } from 'constructs'
+
+const app = new cdk.App({
+  context: {
+    name: 'github-actions',
+  },
+})
+
+type RepoIsh = `${string}/${string}`
```
Comment (dnys1) [inline cdk/src/app-github-actions.ts:12]: Huh TIL - TypeScript's type system is _wild_

File: cdk/src/app-github-actions.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        url: 'https://token.actions.githubusercontent.com',
+        clientIds: ['sts.amazonaws.com'],
+      }
+    )
+
+    /**
+     * Create a principal for the OpenID; which can allow it to assume
+     * deployment roles.
+     */
+    const GitHubPrincipal = new iam.OpenIdConnectPrincipal(
```
Comment (dnys1) [inline cdk/src/app-github-actions.ts:52]: Super nit. In my repo, I would enforce this, but I don't believe TS has as strong of conventions.

```suggestion
    const gitHubPrincipal = new iam.OpenIdConnectPrincipal(
```

**Review by dnys1 — DISMISSED** [review]
LGTM! 🚀

File: cdk/src/app-github-actions.ts (hunk 2 above)
Comment (josefaidt) [reply cdk/src/app-github-actions.ts:52]: good callout, to keep the convention consistent I'll update this to just `principal`

File: cdk/src/pkg.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { access, readFile } from 'node:fs/promises'
+import { resolve } from 'node:path'
+
+const path = resolve('package.json')
```
Comment (josefaidt) [inline cdk/src/pkg.ts:4]: could possibly pick up `cdk/package.json`, revert to using `fileURLToPath(new URL('......`

File: .github/workflows/release-env.yml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+      - uses: actions/setup-node@v3
+        with:
+          node-version-file: '.nvmrc'
+          cache: 'pnpm'
+      - run: pnpm install --frozen-lockfile
+      - name: cdk synth
+        run: |
+          pnpm --filter ./cdk run synth \
+            --context env=${{ inputs.env }} \
+            --context version=${{ inputs.version }}
```
Comment (josefaidt) [inline .github/workflows/release-env.yml:46]:
```suggestion
            --context version=${{ inputs.version }}
            --quiet
```

File: .github/workflows/release-env.yml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
```
Comment (josefaidt) [inline .github/workflows/release-env.yml:46]:
```suggestion
            --context version=${{ inputs.version }} \
```

**Pull request merged**